ansible_base/rbac/models.py

```diff
@@ -333,7 +330,7 @@ def __init__(self, *args, **kwargs):
         """
         super().__init__(*args, **kwargs)
         if not self.id:
-            user = get_current_user()
+            user = current_user_or_system_user()
             if user:
                 # Hazard: user can be a SimpleLazyObject, so use id
                 self.created_by_id = user.id
```

Comment:
You shouldn't need this. This should be handled by the parent class save methods now. Can't you remove the entire block starting with `if not self.id:`?

Reply:
I thought that, but the doc comment of that method says:

```
        """
        Because through models are created via a bulk_create, the save method is usually not called
        to get around this, we populate the user model after initialization
        """
```

so I figured you knew of a case where we wouldn't end up setting `created_by`.

Reply:
Ah ha. I am pretty sure I changed that.

https://github.com/ansible/django-ansible-base/blob/d45d5a18228d68ca01e7410f207500eb140ef8cd/ansible_base/rbac/models.py#L213

That covers the main API use case. Custom through_fields don't even really work in migrations anyway, so I am not really worried about getting rid of it. I want this to be a direct use of the base immutable model.

Reply:
Ok, I dropped that block from `__init__()`.